src/utils.jl

```diff
@@ -146,3 +146,11 @@ function trycholesky(a::Matrix{Float64})
         return e
     end
 end
+
+# Similarity function, per issue #530 (and by extension, issue #482)
+import Base.similar 
+
+function similar(dist::Distribution, params...)
```

Comment:
No need to `import`; you can extend directly by prepending `Base.` to `similar(...)`.

Also, you can simplify this a bit:
```julia
Base.similar(dist::Distribution, params...) = typeof(T).name.wrapper(params...)
```
Note though that I there seems to be an ambiguity with a deprecated method in Base, so you'll likely need a disambiguating definition as well:
```julia
Base.similar(dist::Distribution, params::Union{Integer,AbstractUnitRange}...) =
    typeof(dist).name.wrapper(params...)
```
We can remove that definition once we require Julia 1.0, since the deprecated method in Base will be removed.

Lastly, we should likely document that this will only work if a `UnionAll` constructor for the given distribution is defined, e.g. `Normal(m, v)` versus `Normal{Float32}(m, v)`. If there are any distributions in this package where that is not the case, we could add specific methods to those, either of `similar` or adding constructors.

Reply:
Good points re `import` and one-lining. RE docs, seems reasonable, although I think this may be more general (e.g., `Normal(m, v)` where `m::T` and `v::T` should be OK?).

Reply:
I mean that for a distribution type `D{T}` where `T` dictates the type of the parameter `x`, this method will only work if a constructor `D(x)` is defined. Note that constructors such as that are _not_ defined by default when defining a type; only `D{T}(x)` is defined by default.

Reply:
Done. 

```
# Similarity function, per issue #530 (and by extension, issue #482)
"""
A function to, given a distribution of type D and parameters A, B, C..., create a new D(A, B, C...).

Works so long as there is a type constructor for the distribution; i.e., Normal for Normal{Float64}. 
"""
Base.similar(dist::Distribution, params::Union{Number, AbstractArray}...) = (typeof(dist).name).wrapper(params...)
```

Reply:
And OK. Didn't know that re default constructors. Thanks.